cynic-codegen/src/input_object_derive/field_serializer.rs

```diff
@@ -59,15 +59,15 @@ impl<'a> FieldSerializer<'a> {
             }
         };
 
-        let mut ty = &self.rust_field.ty;
-        if self.should_auto_skip_serializing() {
-            ty = types::outer_type_is_option(ty).expect(
-                "should_auto_skip_serializing is returning true when outer type is not option",
-            )
-        }
+        let aligned_type = align_input_type(
+            &types::parsing2::parse_rust_type(&self.rust_field.ty),
+            &self.graphql_field.value_type,
+            self.graphql_field.has_default,
+        )
+        .to_syn();
```

Comment:
This could be much nicer - probably want to make the align_x_type functions take a syn::Type and do the parsing themselves